docs/source/hyper-parameter-search.rst

```diff
@@ -179,28 +179,29 @@ tasks in the graph and only perform the fit step once for any
 parameter/data/estimator combination. For pipelines that have relatively
 expensive early steps, this can be a big win when performing a grid search.
 
-.. _hyperparameter.incremental:
+.. _hyperparameter.adaptive:
 
 
-Incremental Hyperparameter Optimization
----------------------------------------
+Adaptive Hyperparameter Optimization
+------------------------------------
 
-The second category of hyperparameter optimization uses *incremental* hyperparameter
+The second category of hyperparameter optimization uses *adaptive* hyperparameter
 optimization.
 
 .. autosummary::
-   dask_ml.model_selection.IncrementalSearch
+   dask_ml.model_selection.DecaySearchCV
 
-Broadly speaking, incremental optimization starts with a batch of models (underlying
-estimators and hyperparameter combinationms) and repeatedly calls the underlying estimator's
-``partial_fit`` method with batches of data.
+
+These model selection algorithms are *adaptive*: they decide to keep training
```

Comment:
I think we should split these two ideas a bit more cleanly in the docs:

1. Adaptive: Using previous fit information to decide which models to train next
2. Incremental: Training on large datasets by incrementally passing batches to `partial_fit`.

As written, this paragraph implies that both of these are necessary, when you could imagine Incremental without Adaptive, and Adaptive without Incremental.

I'm going back a bit on what we discussed yesterday. Our implementation *requires* Incremental (it only works with underlying estimators that implement `partial_fit`). So to me that's the most important of the two, and if we're only going to highlight one (e.g. the section title) it should be incremental.

Reply:
I agree with you; I'm also rethinking naming. I agree that `Incremental` is more important to highlight than `Adaptive`; all adaptive algorithms must be incremental, but not all incremental algorithms are adaptive.

TODO:
- [ ] rename `BaseAdaptiveSearchCV` => `BaseIncrementalSearchCV`